src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -412,16 +577,27 @@ private static void WriteCommentSubBlock(Stream stream, ReadOnlySpan<char> comme
     /// <param name="metadata">The metadata of the image or frame.</param>
     /// <param name="transparencyIndex">The index of the color in the color palette to make transparent.</param>
     /// <param name="stream">The stream to write to.</param>
-    private void WriteGraphicalControlExtension(GifFrameMetadata metadata, int transparencyIndex, Stream stream)
+    private void WriteGraphicalControlExtension(GifFrameMetadata? metadata, int transparencyIndex, Stream stream)
     {
+        bool hasTransparency;
+        if (metadata is null)
+        {
+            metadata = new();
+            hasTransparency = transparencyIndex > -1;
```

Comment:
```suggestion
            hasTransparency = transparencyIndex >= 0;
```
Comparisons with `0` a tad faster and allows the cpu itself to optimize (fuse) the instruction.
As the change is simple and code as readable as the current one, I'd do such a change.

Reply:
Done